client/src/pages/HomePage.jsx

```diff
@@ -1,26 +1,99 @@
-import FeaturedBlog from '../components/FeaturedBlog';
-import TrendingBlogs from '../components/TrendingBlogs';
-import { BlogContext } from '../context/BlogContext';
-import { useContext } from 'react';
-import Loader from '../components/Loader'
-import WriteButton from '../components/WriteButton';
+import { Link } from "react-router-dom";
+import { useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { getAllBlogs } from "../features/blogs/blogsSlice";
+import Loader from "../components/Loader";
+import WriteBlogButton from "../components/WriteBlogButton";
+import BlogCard from "../components/BlogCard";
+import moment from "moment";
 
 const HomePage = () => {
-  const { blogs } = useContext(BlogContext);
-  const sortedBlogs = blogs.sort((a, b) => b.impressions - a.impressions);
+  const dispatch = useDispatch();
+  const blogs = useSelector((state) => state.blogs.blogs);
 
-  const featuredBlog = sortedBlogs.length > 0 ? blogs[0] : null;
-  const trendingBlogs = sortedBlogs.length > 1 ? blogs.slice(1, 5) : [];
+  useEffect(() => {
+    if (blogs.length === 0) {
+      dispatch(getAllBlogs());
+    }
+  }, [blogs.length, dispatch]);
 
-  if (sortedBlogs.length === 0) {
-    return <Loader/>
+  if (blogs.length === 0) {
+    return <Loader />;
   }
 
+  const sortedBlogs = [...blogs].sort((a, b) => b.impressions - a.impressions);
+  const featuredBlog = sortedBlogs.length > 0 ? sortedBlogs[0] : null;
+  const trendingBlogs = sortedBlogs.length > 1 ? sortedBlogs.slice(1, 5) : [];
+
+  const formattedDate = moment(featuredBlog.date).format("MMMM d, YYYY");
+
   return (
-    <div className='min-h-screen'>
-      <FeaturedBlog blog={featuredBlog} />
-      <TrendingBlogs blogs={trendingBlogs} />
-      <WriteButton/>
+    <div className="min-h-screen">
+      <section className="container p-6 mx-auto space-y-6 sm:space-y-12 my-6">
+        <div className="mb-6 text-left">
+          <h2 className="text-3xl font-bold">Featured Blog</h2>
+        </div>
+        {featuredBlog && (
+          <Link key={featuredBlog._id} to={`/blogs/${featuredBlog._id}`}>
+            <div className="block gap-3 mx-auto sm:max-w-full group lg:grid lg:grid-cols-12">
+              <img
+                src={featuredBlog.imageUrl}
+                alt=""
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Missing alt text for featured image</b></div>
<div id="fix">

Consider adding a descriptive `alt` text for the featured blog image to improve accessibility.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```suggestion
                alt={featuredBlog.title}
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/Divyanshu9822/blog-mern-app/pull/22#issuecomment-2544587479>#245782</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged